### PR TITLE
Public internet test

### PR DESCRIPTION
**Pull request by kklin**

### CODE REVIEW

File: specs/nginx/Makefile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+REPO = quilt/nginx
```
Comment (kklin) [inline specs/nginx/Makefile]: This image needs to be pushed for this spec to work.

File: specs/nginx/default.tmpl [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    location / {
+        root   /usr/share/nginx/html;
+        index  index.html index.htm;
+    }
+
+    # redirect server error pages to the static page /50x.html
+    #
+    error_page   500 502 503 504  /50x.html;
+    location = /50x.html {
```
Comment (LuiseV) [inline specs/nginx/default.tmpl]: Is it necessary to set this location explicitly? The root is the same as we set above, and the path prefix matches.

File: quilt-tester/tests/pub-facing/check_public_connections.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+		log.Info("PASSED")
+	} else {
+		log.Info("FAILED")
+	}
+}
+
+func test(machines []db.Machine, containers []db.Container,
+	connections []db.Connection) bool {
+
+	// Map of label to publicly-connected ports to the label.
```
Comment (LuiseV) [inline quilt-tester/tests/pub-facing/check_public_connections.go]: Maybe "Map of label to its publicly exposed ports" or something like that. I had to read this comment a few times.

File: specs/nginx/default.tmpl (hunk 2 above)
Comment (kklin) [inline specs/nginx/default.tmpl]: The config is copied from the official nginx container, so I'm inclined to just keep it the same. Does that sound alright?

File: quilt-tester/tests/pub-facing/check_public_connections.go (hunk 3 above)
Comment (kklin) [inline quilt-tester/tests/pub-facing/check_public_connections.go]: Fixed, that sounds much better 😄

File: specs/nginx/default.tmpl (hunk 2 above)
Comment (LuiseV) [inline specs/nginx/default.tmpl]: Sure, then just keep it. But I wonder why they wrote it that way. Exact location matches are apparently faster, but I would think that has practically no effect here.... But sounds good, I'm good with this then.

**Pull request merged**